### ISSUE TITLE
yarn install link step fails on Windows 10

### ISSUE THREAD

**Issue by CharlesKiarie** [medusajs/medusa#3610]
### Describe the bug

Running `yarn install` after cloning the repo fails with the error below.

```
➤ YN0000: ┌ Link step
➤ YN0001: │ Error: While cloning /C:/Development/medusa/packages/stock-location/node_modules/jest-get-type -> /C:/Development/medusa/node_modules/jest-jasmine2/node_modules/jest-get-type UNKNOWN: unknown error, link 'C:\Development\medusa\packages\stock-location\node_modules\jest-get-type\LICENSE' -> 'C:\Development\medus\node_modules\jest-jasmine2\node_modules\jest-get-type\LICENSE'
```

### System information

Medusa version (including plugins): 1.7.15
Node.js version: 16.14.2
Yarn version: 3.2.1
Operating system: Windows 10

### Steps to reproduce the behavior

1. Clone repo
2. Run `yarn install`

### Additional context

This bug has been submitted and discussed multiple times here [#2703](https://github.com/medusajs/medusa/issues/2703) and here [#2181](https://github.com/medusajs/medusa/issues/2181) a commenter gave a solution that has worked for me here [cf1331](https://github.com/medusajs/medusa/issues/2181#issuecomment-1284075932) which is change `nmMode: hardlinks-global` to `nmMode: hardlinks-local` in `.yarnrc.yml` and running `yarn install`

**Comment by CharlesKiarie** [comment]
I will be doing a brief risk assessment on whether this change should be officially added to medusa js and how does it affect the project both going over positives and negatives before submitting a pull request. My goal is to make the developer experience for developers looking to contribute to medusa friendly and straightforward.

**Comment by carlos-r-l-rodrigues** [comment]
you can update the `nmMode` in your `.yarnrc.yml` file to `nmMode: classic`

**Comment by CharlesKiarie** [comment]
Using hardlinks-global Medusa's size on disk was 2.72gb
Using hardlinks-local Medusa's size on disk was 3.18gb

I tried various ways to get the project to yarn install on v3.2.1  `nmMode: hardlinks-global` it still failed during the linking step. I tried to use Yarn classic(v1.22.19) in which yarn install worked but just caused a lot of gremlins with other commands not being recognized.

So far my only solution that just works has been changing `nmMode: hardlinks-local` in .yarnrc.yml.
PROS:
- Only have to keep track of one change easy to stash and to ignore said change during a rebase
- Medusa now works
- Using up to date dependencies similar to the development team
CONS:
- Size on disk is a lot larger
- I still have to remind myself of the change I made so as not to add it to a pr


CONCLUSION:
- So far everyone suffering from this issue has found a different way to solve it I wanted to see if a de facto fix existed and I am don't think a pr to change .yarnrc.yml is the best approach as it seems majority of active contributors have no issues with `nmMode: hardlinks-global`.
- I am also yet to come across a good explanation to why linking step fails on some windows 10 systems.
- Will look into it more on the Yarn discord and see if anyone can help until then I am closing this thread.
